torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -170,10 +170,20 @@ static void setInputBroadcastGroups(KernelSpec& spec) {
 // group outputs when several outputs had the same calculation except
 // for the final _grad_sum_to_size. This is also the reason why
 // we need to deduplicate kernel outputs at the end of this function.
+// For FusedConcats we actually need to record the SumtoSize per concat-input.
+// (which is why for each output we have a vector of indices).
+// we use that FusedConcats come directly before the results.
 void processGradSumToSize(KernelSpec& spec) {
   auto graph = spec.graph();
 
-  std::vector<int64_t> outputGradSumToSizes(graph->outputs().size(), -1);
+  std::vector<std::vector<int64_t>> outputGradSumToSizes(
```

Comment:
Why is the a vector of vectors now?? Any comments on what do we actually record?

Reply:
So the comment three lines above
```
// For FusedConcats we actually need to record the SumtoSize per concat-input.
// (which is why for each output we have a vector of indices).
// we use that FusedConcats come directly before the results.
```
doesn't say? (And the old bits for what we collect.)